hoomd/mpcd/SphereGeometry.h

```diff
@@ -85,23 +89,28 @@ class __attribute__((visibility("default"))) SphereGeometry
              * The perpendicular and parallel components of the velocity are:
              * v_perp = (v.n)n = (v.r/R^2)r
              * v_para = v-v_perp
+	     * V_vec is vector component of V(interface velocity)
              */
             if (m_bc == boundary::no_slip)
                 {
-                /* No-slip and no penetration requires reflection of both parallel and perpendicular components.
-                 * This results in just flipping of all the velocity components.
+                /* No-slip and no penetration requires reflection of both parallel component and perpendicular(relative to interface) component
+		 * V_perp(new) = -(V_perp(old)-V_interface)+V_interface 
+                 * V_para(new) = -V_para(old)
+                 * This results in just V_new = - v_old + 2*V_interface. 
                  */
-                vel = -vel;
+		const Scalar3 V_vec = m_V*pos/m_R;
```

Comment:
To reduce code duplication, put this line before the `if-else if` since it is used in both branches.